main.go

```diff
@@ -165,19 +233,75 @@ func genMessage(g *protogen.GeneratedFile, m *protogen.Message) {
 		defer genOneofEnum(g, m, oneof)
 	}
 
-	g.P("} derive(Default, Eq, Show)")
+	g.P("} derive(Eq, Show)")
 
 	genMessageSize(g, m)
 	genMessageRead(g, m)
 	genMessageWrite(g, m)
+	genMessageDefault(g, m)
+}
+
+func genMessageDefault(g *protogen.GeneratedFile, m *protogen.Message) {
+	fmt.Fprintf(g, "pub impl Default for %s with default() -> %s {\n", m.GoIdent.GoName, m.GoIdent.GoName)
+	fmt.Fprintf(g, "  {\n")
+
+	// Initialize all fields to their default values
+	for _, field := range m.Fields {
+
+		fieldName := PascalToSnake(field.Desc.JSONName())
+		if field.Oneof != nil && field.Oneof.Desc.IsSynthetic() {
+			fmt.Fprintf(g, "    %s: None,\n", fieldName)
+		} else if field.Desc.HasOptionalKeyword() {
+			if field.Desc.HasDefault() {
+				switch field.Desc.Kind() {
+				case protoreflect.BoolKind:
+					fmt.Fprintf(g, "    %s: Some(%t),\n", fieldName, field.Desc.Default().Bool())
+				case protoreflect.Int32Kind, protoreflect.Sfixed32Kind, protoreflect.Sint32Kind, protoreflect.Uint32Kind:
+					fmt.Fprintf(g, "    %s: Some(%d),\n", fieldName, field.Desc.Default().Int())
+				case protoreflect.Int64Kind, protoreflect.Sfixed64Kind, protoreflect.Sint64Kind, protoreflect.Uint64Kind:
+					fmt.Fprintf(g, "    %s: Some(%d),\n", fieldName, field.Desc.Default().Int())
+				case protoreflect.FloatKind:
+					fmt.Fprintf(g, "    %s: Some(%f),\n", fieldName, field.Desc.Default().Float())
+				case protoreflect.DoubleKind:
+					fmt.Fprintf(g, "    %s: Some(%f),\n", fieldName, field.Desc.Default().Float())
+				case protoreflect.StringKind:
+					fmt.Fprintf(g, "    %s: Some(\"%s\"),\n", fieldName, field.Desc.Default().String())
+				case protoreflect.BytesKind:
+					fmt.Fprintf(g, "    %s: Some(b\"%s\"),\n", fieldName, field.Desc.Default().Bytes())
+				case protoreflect.EnumKind:
+					fmt.Fprintf(g, "    %s: Some(%s::from_enum(%v)),\n", fieldName, field.Enum.GoIdent.GoName, field.Desc.Default().Enum())
+				}
+			} else {
+				fmt.Fprintf(g, "    %s: None,\n", fieldName)
+			}
+		} else if field.Desc.IsList() {
+			fmt.Fprintf(g, "    %s: Array::new(),\n", fieldName)
+		} else {
+			defaultValue := getMbtType(field)
+			fmt.Fprintf(g, "    %s: %s::default(),\n", fieldName, defaultValue)
```

Comment:
```suggestion
			fmt.Fprintf(g, "    %s: Default::default(),\n", fieldName)
```
Maybe this is enough?